### PR TITLE
Fetch app logs in DISTRIBUTED context when possible.

### PR DESCRIPTION
**Pull request by wmarshall484**

### CODE REVIEW

**Comment by wmarshall484** [comment]
Needs to be tested, and the warning message could be improved for when `always_fetch_logs` is set on a DISTRIBUTED submission that doesn't support fetching logs.

**Comment by ddebrunner** [comment]
The docstrings for the retrieve methods should document that `None` can be returned.

File: com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py [hunk 1]
```diff
@@ -561,8 +561,10 @@ def retrieve_log_trace(self, filename=None, dir=None):
 
         if not filename:
             filename = _file_name('job', self.id, '.tar.gz')
- 
-        return self.rest_client._retrieve_file(self.applicationLogTrace, filename, dir, 'application/x-compressed')
+        if hasattr(self, "applicationLogTrace") and self.applicationLogTrace is not None:
```
Comment (ddebrunner) [inline com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py]: Ideally the creating a default `filename` would occur only if the method is going to retrieve logs.

**Comment by ddebrunner** [comment]
Multiple methods use `_retrieve_file`, they should all have similar checks.

**Pull request merged**